### PR TITLE
Translator is null, if we dont use a translator

### PR DESCRIPTION
**Pull request by kokspflanze**
hi,

the main problem was [Translator must be an instanceof "Zend\I18n\Translator\Translator" or "Zend\I18n\Translator\TranslatorInterface" @ thadafinser/zfc-datagrid/src/ZfcDatagrid/Renderer/AbstractRenderer.php:391], to fix the problem i added the hasTranslation check @ getRenderer and loadData.

ChangeLog for ZfcDatagrid/Datagrid

- added some php-doc-blocks
- changed count to bool for better performance
- added hasTranslation validation @ getRenderer && loadData, if we dont have a translator

### CODE REVIEW

**Comment by ThaDafinser** [comment]
@kokspflanze can you please rebase your PR, there are merge conflicts.

Can you also run `php-cs-fixer fix` to fix the standards problem?

**Comment by kokspflanze** [comment]
sure, i will create a new PR in the evening, to fix the problems =)

File: src/ZfcDatagrid/Datagrid.php [hunk 1]
```diff
@@ -749,11 +760,7 @@ public function getRowStyles()
      */
     public function hasRowStyles()
     {
-        if (count($this->rowStyles) > 0) {
-            return true;
-        }
-
-        return false;
+        return (bool) $this->rowStyles;
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/Datagrid.php]: @kokspflanze this is shorter, but i (currently/personally) dont like it.
It's harder to understand IMO (someone who comes from another language wont know such a special feature).

But no worry, i'll merge it still...it seems to work at least

Comment (kokspflanze) [inline src/ZfcDatagrid/Datagrid.php]: it is the fastest way to check if its empty or not, !empty($foo), is also possible but a bit slower and count($foo) is the slowest way in this case. So for best performance i changed it to cast in bool.

**Comment by kokspflanze** [comment]
new PR is open https://github.com/ThaDafinser/ZfcDatagrid/pull/172

**Comment by stevenbuehner** [comment]
I guess the problem still exists, if no translator is used. Because the renderes don't check if the translator is set. They REQUIRE one.

For Example the ExcelRenderer:
```
        // Sheet 1
        $phpExcel->setActiveSheetIndex(0);
        $sheet = $phpExcel->getActiveSheet();
        $sheet->setTitle($this->getTranslator()
            ->translate($optionsRenderer['sheetName']));```

This means, that you will get a "Call to a member function translate() on null" Error, when no translator has been set yet. Same with CSV, ...

**Comment by kokspflanze** [comment]
thanks @stevenbuehner i will check it today and search all parts and fix them

**Pull request closed without merging**